openfisca_france/model/prelevements_obligatoires/impot_revenu/ir.py

```diff
@@ -1108,6 +1108,33 @@ def formula(foyer_fiscal, period, parameters):
         return (taux_effectif == 0) * nbptr * bareme.calc(rni / nbptr) + taux_effectif * rni
 
 
+class ir_taux_marginal(Variable):
+    value_type = float
+    entity = FoyerFiscal
+    label = "Taux marginal de l'impôt sur le revenu"
+    definition_period = YEAR
+
+    def formula(foyer_fiscal, period, parameters):
+        nbptr = foyer_fiscal('nbptr', period)
+        taux_effectif = foyer_fiscal('taux_effectif', period)
+        rni = foyer_fiscal('rni', period)
+        bareme = parameters(period).impot_revenu.bareme
+        return (taux_effectif == 0) * bareme.marginal_rates(rni / nbptr) + taux_effectif
+
+
+class ir_tranche(Variable):
+    value_type = int
+    entity = FoyerFiscal
+    label = "Tranche du barème appliquée"
+    definition_period = YEAR
```

Comment:
Une autre proposition, or je ne sais pas si c'est la meilleure référence disponible 

```suggestion
    reference = "https://impots.dispofi.fr/bareme-impot/calcul-impot-par-tranche"
    definition_period = YEAR
```